src/zarr/api/asynchronous.py

```diff
@@ -194,11 +195,16 @@ async def consolidate_metadata(
     -------
     group: AsyncGroup
         The group, with the ``consolidated_metadata`` field set to include
-        the metadata of each child node.
+        the metadata of each child node. If the Store doesn't prefer
+        consolidated metadata, this is function does nothing and returns
+        the group without modifications. See ``Store.supports_consolidated_metadata``.
     """
     store_path = await make_store_path(store, path=path)
 
     group = await AsyncGroup.open(store_path, zarr_format=zarr_format, use_consolidated=False)
+    if not group.store_path.store.supports_consolidated_metadata:
```

Comment:
```suggestion
    if not store.supports_consolidated_metadata:
```

Reply:
I cannot use `store` directly because it's a StoreLike, but I can certainly use `store_path.store`. Will change